src/Evolution/Systems/GeneralizedHarmonic/Tags.hpp

```diff
@@ -61,5 +62,42 @@ struct SpacetimeDerivGaugeH : db::SimpleTag {
   using type = tnsr::ab<DataVector, Dim, Frame>;
   static std::string name() noexcept { return "SpacetimeDerivGaugeH"; }
 };
+
+// {@
+/// \ingroup GeneralizedHarmonicGroup
+/// \brief Tags corresponding to the characteristic fields of the generalized
+/// system.
+///
+/// \details For details on how these are defined and computed, see
+/// ComputeCharacteristicSpeeds
+template<size_t Dim, typename Frame>
+struct UPsi {
+  using type = tnsr::aa<DataVector, Dim, Frame>;
+  static std::string name() noexcept { return "UPsi"; }
+};
+template<size_t Dim, typename Frame>
+struct UZero {
+  using type = tnsr::iaa<DataVector, Dim, Frame>;
+  static std::string name() noexcept { return "UZero"; }
+};
+template<size_t Dim, typename Frame>
+struct UPlus {
+  using type = tnsr::aa<DataVector, Dim, Frame>;
+  static std::string name() noexcept { return "UPlus"; }
+};
+template<size_t Dim, typename Frame>
+struct UMinus {
+  using type = tnsr::aa<DataVector, Dim, Frame>;
+  static std::string name() noexcept { return "UMinus"; }
+};
+// @}
+
+template<size_t Dim, typename Frame>
+struct CharacteristicSpeeds : db::SimpleTag {
+  using type = Variables<db::wrap_tags_in<
```

Comment:
The reasoning was you can only defined characteristic fields for symmetric hyperbolic systems, 
and so for these systems the characteristics are thought about as "a set of fields each of which has a corresponding speed", whereas for hydro, the characteristics are matrices of left and right eigenvectors, and the speeds are the eigenvalues. The variables makes sense here because the fields and speeds are named and correspond to tags well. For implementing upwind flux it will then be useful to loop over the tags and get the corresponding speed and field. 

As to whether ValenciaDivClean should be changed, I think an array is perfectly fine to represent the speeds, but that doesn't seem like something this PR should be concerned with.

Reply:
I don't see what symmetric hyperbolicity has to do with that. I think that strong hyperbolicity should always suffice to define characteristic fields.

Reply:
We still think that Valencia should be changed to a Variables when able, but we agree with @osheamonn that it shouldn't be in this PR.